packages/google_maps_flutter/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -524,98 +527,27 @@ class _GoogleMapState extends State<GoogleMap> {
-        minMaxZoomPreference = map.minMaxZoomPreference,
-        rotateGesturesEnabled = map.rotateGesturesEnabled,
-        scrollGesturesEnabled = map.scrollGesturesEnabled,
-        tiltGesturesEnabled = map.tiltGesturesEnabled,
-        trackCameraPosition = map.onCameraMove != null,
-        zoomControlsEnabled = map.zoomControlsEnabled,
-        zoomGesturesEnabled = map.zoomGesturesEnabled,
-        liteModeEnabled = map.liteModeEnabled,
-        myLocationEnabled = map.myLocationEnabled,
-        myLocationButtonEnabled = map.myLocationButtonEnabled,
-        padding = map.padding,
-        indoorViewEnabled = map.indoorViewEnabled,
-        trafficEnabled = map.trafficEnabled,
-        buildingsEnabled = map.buildingsEnabled,
-        assert(!map.liteModeEnabled || Platform.isAndroid);
-
-  final bool compassEnabled;
-
-  final bool mapToolbarEnabled;
-
-  final CameraTargetBounds cameraTargetBounds;
-
-  final MapType mapType;
-
-  final MinMaxZoomPreference minMaxZoomPreference;
-
-  final bool rotateGesturesEnabled;
-
-  final bool scrollGesturesEnabled;
-
-  final bool tiltGesturesEnabled;
-
-  final bool trackCameraPosition;
-
-  final bool zoomControlsEnabled;
-
-  final bool zoomGesturesEnabled;
-
-  final bool liteModeEnabled;
-
-  final bool myLocationEnabled;
-
-  final bool myLocationButtonEnabled;
-
-  final EdgeInsets padding;
-
-  final bool indoorViewEnabled;
-
-  final bool trafficEnabled;
-
-  final bool buildingsEnabled;
-
-  Map<String, dynamic> toMap() {
-    return <String, dynamic>{
-      'compassEnabled': compassEnabled,
-      'mapToolbarEnabled': mapToolbarEnabled,
-      'cameraTargetBounds': cameraTargetBounds.toJson(),
-      'mapType': mapType.index,
-      'minMaxZoomPreference': minMaxZoomPreference.toJson(),
-      'rotateGesturesEnabled': rotateGesturesEnabled,
-      'scrollGesturesEnabled': scrollGesturesEnabled,
-      'tiltGesturesEnabled': tiltGesturesEnabled,
-      'zoomControlsEnabled': zoomControlsEnabled,
-      'zoomGesturesEnabled': zoomGesturesEnabled,
-      'liteModeEnabled': liteModeEnabled,
-      'trackCameraPosition': trackCameraPosition,
-      'myLocationEnabled': myLocationEnabled,
-      'myLocationButtonEnabled': myLocationButtonEnabled,
-      'padding': <double>[
-        padding.top,
-        padding.left,
-        padding.bottom,
-        padding.right,
-      ],
-      'indoorEnabled': indoorViewEnabled,
-      'trafficEnabled': trafficEnabled,
-      'buildingsEnabled': buildingsEnabled,
-    };
-  }
-
-  Map<String, dynamic> updatesMap(_GoogleMapOptions newOptions) {
-    final Map<String, dynamic> prevOptionsMap = toMap();
-
-    return newOptions.toMap()
-      ..removeWhere(
-          (String key, dynamic value) => prevOptionsMap[key] == value);
-  }
+/// Builds a [MapConfiguration] from the given [map].
+MapConfiguration _configurationFromMapWidget(GoogleMap map) {
+  assert(!map.liteModeEnabled || Platform.isAndroid);
+  return MapConfiguration(
```

Comment:
> Maybe add a `GoogleMap.cloneMapConfiguration` method that returns `MapConfiguration()..diff(_mapConfiguration);`
> 
> That way you don't have to repeat the cloning of MapConfiguration in this helper method.

I'm not following; `_configurationFromMapWidget` is what *creates* `_mapConfiguration` (from the separate-but-mostly-identical widget properties).

> (Is [this](https://github.com/flutter/plugins/blob/main/packages/google_maps_flutter/google_maps_flutter/lib/src/controller.dart#L55-L68) the only need for public getters for the widget values?)

I'm not sure, but since `GoogleMap` is public, we can't change it without a breaking change to the whole plugin.

Reply:
Ah, yes, I misunderstood this bit; I thought that the `GoogleMap` widget was created with the new `MapConfiguration` objects too... I guess I thought we had a `GoogleMap.fromConfiguration` constructor, and not all the possible little values exploded in the GoogleMap widget instance! (yikes).